File: src-tauri/src/models/recording/recording.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SessionRecording {
    pub id: String,
    pub terminal_id: String,
    pub session_name: String,
    pub terminal_type: String, // "Local" | "SSH"
    pub started_at: DateTime<Utc>,
    pub ended_at: Option<DateTime<Utc>>,
    pub duration_ms: Option<i64>,
    pub file_path: String,
    pub file_size: i64,
    pub width: u16,
    pub height: u16,
    pub metadata: Option<String>, // JSON
    pub created_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AsciicastHeader {
    pub version: u8,
    pub width: u16,
    pub height: u16,
    pub timestamp: Option<i64>,
    pub title: Option<String>,
    pub env: Option<serde_json::Value>,
}

File: src/types/recording.ts
export interface SessionRecording {
  id: string;
  terminalId: string;
  sessionName: string;
  terminalType: "Local" | "SSH";
  startedAt: string;
  endedAt?: string;
  durationMs?: number;
  filePath: string;
  fileSize: number;
  width: number;
  height: number;
  metadata?: string;
  createdAt: string;
}

export interface RecordingMetadata {
  tags: string[];
  description?: string;
  sshHost?: string;
  shell?: string;
}

export interface AsciicastHeader {
  version: number;
  width: number;
  height: number;
  timestamp?: number;
  title?: string;
  env?: Record<string, string>;
}

export interface AsciicastEvent {
  time: number;
  eventType: string;
  data: string;
}
